containers/ReferenceValue.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { changeReferenceValue } from '../actions'
+
+let ReferenceValue = ({ entityId, refId, referenceValue, dispatch }) => (
+  <div>
+    <input type="text" 
+      placeholder="reference value"
+      value={referenceValue}
+      onChange={e => dispatch(changeReferenceValue(entityId, refId, e.target.value))}/>
```

Comment:
Got it. 
For component and container issue, those different actions can still pass from parent, so that only EntityPanel be container and it get all actions and pass down